Library/OcBootManagementLib/OcBootManagementLib.c

```diff
@@ -329,14 +329,11 @@ OcRunBootPicker (
         }
 
         //
-        // Do screen clearing for builtin menu here, so that it is possible to see the action.
-        // TODO: Probably remove that completely.
+        // Clear screen from previous console contents (builtin picker,
+        // or previous console tool even in Canopy) before loading the entry.
         //
+        gST->ConOut->ClearScreen (gST->ConOut);
         if (Context->ShowMenu == OcShowSimpleBootMenu) {
-          //
-          // Clear screen from picker contents before loading the entry.
-          //
-          gST->ConOut->ClearScreen (gST->ConOut);
           gST->ConOut->TestString (gST->ConOut, OC_CONSOLE_MARK_UNCONTROLLED);
```

Comment:
I am less certain that having this (here, and only when using builtin picker) is correct, but I think so?

Reply:
Decision to leave for now, since that doesn't change the order of operations for builtin (which were optimised to avoid flashes etc.) and does appear to work fine on Canopy - at least on my system.